enterprise/wsproxy/wsproxy_test.go

```diff
@@ -498,3 +600,86 @@ func TestWorkspaceProxyWorkspaceApps(t *testing.T) {
 		}
 	})
 }
+
+// createDERPClient creates a DERP client and spawns a goroutine that reads from
+// the client and sends the received packets to a channel.
+func createDERPClient(t *testing.T, name string, derpURL string) (*derphttp.Client, <-chan derp.ReceivedPacket) {
+	t.Helper()
+
+	client, err := derphttp.NewClient(key.NewNode(), derpURL, func(format string, args ...any) {
+		t.Logf(name+": "+format, args...)
+	})
+	require.NoError(t, err, "create client")
+	t.Cleanup(func() {
+		_ = client.Close()
+	})
+	err = client.Connect(testutil.Context(t, testutil.WaitLong))
```

Comment:
I think it's cleaner to pass in a context.  That way you can set a deadline for the whole test to complete, rather than multiple long timeouts for each step.